tq/transfer_queue_test.go

```diff
@@ -31,6 +44,36 @@ func TestRetryCounterCanNotRetryAfterExceedingRetryCount(t *testing.T) {
 	assert.False(t, canRetry)
 }
 
+func TestRetryCounterDoesNotDelayFirstAttempt(t *testing.T) {
+	rc := newRetryCounter()
+	assert.Equal(t, time.Time{}, rc.ReadyTime("oid"))
+}
+
+func TestRetryCounterDelaysExponentially(t *testing.T) {
+	rc := newRetryCounter()
+	start := time.Now()
+
+	rc.Increment("oid")
+	ready1 := rc.ReadyTime("oid")
+	assert.True(t, ready1.Sub(start) >= baseRetryDelayMs*time.Millisecond)
+
+	rc.Increment("oid")
+	ready2 := rc.ReadyTime("oid")
+	assert.True(t, ready2.Sub(start) >= 2*baseRetryDelayMs*time.Millisecond)
```

Comment:
Can we use `assert.GreaterOrEqual` here and above so that we get a better error message on failure?

Reply:
It's not in the version of `stretchr/testify` that's currently vendored, but I can updated that.